doc/developer/release-checklist.md

```diff
@@ -1,30 +1,99 @@
-# Release Tasks
+# Release checklist
 
-## Code Changes
+## Release candidate
 
-- [ ] update the version field in [`src/materialized/Cargo.toml`](../../src/materialized/Cargo.toml)
+- [ ] Choose the desired commit.
 
-## Verification
+  ```shell
+  git checkout <SHA>
+  ```
 
-- [ ] load/soak tests run on -rc tagged commit
-  - command should be:
-    ```
-    dc.sh clean-load-test
+- [ ] Choose the name for the release candidate following the format
+  `v<VERSION>-rc<N>`, where _N_ starts at 1. For example, the first RC for
+  v0.2.3 would be called v0.2.3-rc1.
+
+- [ ] Update the version field in [`src/materialized/Cargo.toml`](../../src/materialized/Cargo.toml)
+      and commit that change.
```

Comment:
maybe add an "e.g. `version = "0.2.3-rc1"`" to make it super clear that the -rc` shows up in the cargo.toml?

Reply:
done